respy/shared.py

```diff
@@ -489,3 +472,89 @@ def convert_dictionary_keys_to_dense_indices(dictionary):
         new_dictionary[new_key] = val
 
     return new_dictionary
+
+
+def subset_to_period(object, dense_index_to_complex, period):
+    period_keys = [x for x, y in dense_index_to_complex.items() if y[0][0] == period]
+    out = {key: object[key] for key in object.keys() if key in period_keys}
+    return out
+
+
+def subset_to_choice(matrix, choice_set):
+    delete = [i for i, x in enumerate(choice_set) if bool(x) is False]
+    out = np.delete(np.delete(matrix, delete, 0), delete, 1)
+    return out
+
+
+def return_valid_choices(choice_set, optim_paras):
+    return [
+        x for i, x in enumerate(optim_paras["choices"]) if bool(choice_set[i]) is True
```

Comment:
You do not compare to True with `if cond is True`, but simply `if cond`. I do not think you need the bool conversion.